lib/charms/saml_integrator/v0/saml.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Canonical Ltd.
+# Licensed under the Apache2.0. See LICENSE file in charm source for details.
+
+"""Library to manage the relation data for the SAML Integrator charm.
+
+This library contains the SamlRelatioSamlDataAvailableEvent class to encapsulate the relation
+data, providing an improved user experience for the requirer charms developers.
+"""
+
+# The unique Charmhub library identifier, never change it
+LIBID = "511cdfa7de3d43568bf9b512f9c9f89d"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+PYDEPS = ["ops>=2.0.0", "pydantic==1.10.10"]
+
+# pylint: disable=wrong-import-position
+import re
+from typing import Dict, List, Optional
+
+import ops
+from pydantic import AnyHttpUrl, BaseModel, Field
+from pydantic.tools import parse_obj_as
+
+DEFAULT_RELATION_NAME = "saml"
+
+
+class SamlEndpoint(BaseModel):
+    """Represent a SAML endpoint.
+
+    Attrs:
+        name: Endpoint name.
+        url: Endpoint URL.
+        binding: Endpoint binding.
+        response_url: URL to address the response to.
+    """
+
+    name: str = Field(..., min_length=1)
+    url: AnyHttpUrl
+    binding: str = Field(..., min_length=1)
+    response_url: Optional[AnyHttpUrl]
+
+    def to_relation_data(self) -> Dict[str, str]:
+        """Convert an instance of SamlEndpoint to the relation representation.
+
+        Returns:
+            Dict containing the representation.
+        """
+        result: Dict[str, str] = {}
+        http_method = self.binding.split(":")[-1].split("-")[-1].lower()
+        lowercase_name = re.sub(r"(?<!^)(?=[A-Z])", "_", self.name).lower()
```

Comment:
Should we include a comment what this is doing?

Reply:
Done